internal/pkg/vaultworker/password.go

```
@@ -0,0 +1,112 @@
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dghubble/sling"
+	"github.com/sethvargo/go-password/password"
+)
+
+type UserPasswd struct {
+    User string
+    Passwd string
+}
+
+func CreateCredential() (string, error) {
+	pass, err := password.Generate(8, 4, 4, false, false)
+	if err != nil {
+		return "", err
+	}
+	return pass, nil
+}
+
+func CredentialInStore(config *tomlConfig, secretBaseURL string, credPath string, c *http.Client) (bool, error) {
+
+	t, err := GetSecret(config.SecretService.TokenFolderPath + "/" + config.SecretService.VaultInitParm)
+	if err != nil {
+		lc.Error(err.Error())
+		return false, err
+	}
+
+	s := sling.New().Set(VaultToken, t.Token)
+
+	req, err := s.New().Base(secretBaseURL).Get(credPath).Request()
+	resp, err := c.Do(req)
+	if err != nil {
+		errStr := fmt.Sprintf("Failed to retrieve credentials with path as %s with error %s", credPath, err.Error())
+		return false, errors.New(errStr)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		lc.Info(fmt.Sprintf("No credential path found, setting up the credentials for %s", credPath))
+		return false, nil
+	}
+
+	lc.Info(fmt.Sprintf("%s - %d", credPath, resp.StatusCode))
+
+	var result map[string]interface{}
+
+	by, _ := ioutil.ReadAll(resp.Body)
+
+	json.Unmarshal(by, &result)
+
+	credentials := result["data"].(map[string]interface{})
+
+	if len(credentials) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func InitCredentials(config *tomlConfig, secretBaseURL string, secretPath string, cred *UserPasswd, c *http.Client) error {
+	
+	t, err := GetSecret(config.SecretService.TokenFolderPath + "/" + config.SecretService.VaultInitParm)
+	if err != nil {
+		lc.Error(err.Error())
+		return err
+	}
+
+	s := sling.New().Set(VaultToken, t.Token)
+
+	lc.Info("Trying to upload init credentials to secret service server.")
+	req, err := s.New().Base(secretBaseURL).Post(secretPath).BodyJSON(cred).Request()
```

err is not checked here